.github/workflows/main.yml

```diff
@@ -1,72 +1,140 @@
+              TAGS="${TAGS},${IMAGE}:latest"
+              VERSION="${VERSION#v}"
+            fi
+
+          # Has this workflow been triggered by a branch ?
+          elif [[ "${GITHUB_REF}" == refs/heads/* ]]; then
+            VERSION=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+          fi
+
+          if [[ ! -z "${VERSION}" ]]; then
+            TAGS="${TAGS},${IMAGE}:${VERSION}"
+          fi
+
+          echo ::set-output name=version::${VERSION}
+          echo ::set-output name=tags::${TAGS}
+          echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+      - uses: actions/checkout@v2
+      - uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+      - uses: docker/build-push-action@v2
+        with:
+          file: ./Dockerfile
+          build-args: |
+            VERSION_NAME=${{ steps.metadata.outputs.version }}
+            VERSION_SHA=${GITHUB_SHA::8}
+          platforms: linux/amd64
+          outputs: type=docker,dest=image
+          tags: ${{ steps.metadata.outputs.tags }}
+          labels: |
+            com.arkhn.image.title=${{ github.event.repository.name }}
+            com.arkhn.image.description=${{ github.event.repository.description }}
+            com.arkhn.image.url=${{ github.event.repository.html_url }}
+            com.arkhn.image.source=${{ github.event.repository.clone_url }}
+            com.arkhn.image.version=${{ steps.metadata.outputs.version }}
+            com.arkhn.image.created=${{ steps.metadata.outputs.created }}
+            com.arkhn.image.revision=${{ github.sha }}
+            com.arkhn.image.licenses=${{ github.event.repository.license.spdx_id }}
+      - uses: actions/upload-artifact@v2
+        with:
+          name: image
+          path: image
+
+  tests:
     needs: build
-    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: image
+      - run: docker load -i image
+      - run: |
+          IMAGE=$(echo ${{ needs.build.outputs.tags }} | cut -f1 -d,)
+          docker run --entrypoint=pytest ${IMAGE}
+
+  publish-docker:
+    needs: [tests, build]
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: image
+      - run: docker load -i image
+      - uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_LOGIN }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - run: |
+          TAGS=${{ needs.build.outputs.tags }}
+
+          for tag in ${TAGS//,/ }; do
+            docker push "${tag}"
+          done
+
+  publish-pypi:
+    needs: [tests, build]
+    runs-on: ubuntu-18.04
+    if: github.event_name == 'push' && github.event.base_ref == 'refs/heads/master' && startsWith(github.ref, 'refs/tags')
```

Comment:
`github.event_name == 'push'` is redundant since the workflow is only triggered by the push event

Reply:
right